// react-country-flag/dist/index.js
'use strict';

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = require('react');
var React__default = _interopDefault(React);
var PropTypes = _interopDefault(require('prop-types'));

var classCallCheck = function (instance, Constructor) {
  if (!(instance instanceof Constructor)) {
    throw new TypeError("Cannot call a class as a function");
  }
};

var createClass = function () {
  function defineProperties(target, props) {
    for (var i = 0; i < props.length; i++) {
      var descriptor = props[i];
      descriptor.enumerable = descriptor.enumerable || false;
      descriptor.configurable = true;
      if ("value" in descriptor) descriptor.writable = true;
      Object.defineProperty(target, descriptor.key, descriptor);
    }
  }

  return function (Constructor, protoProps, staticProps) {
    if (protoProps) defineProperties(Constructor.prototype, protoProps);
    if (staticProps) defineProperties(Constructor, staticProps);
    return Constructor;
  };
}();

var _extends = Object.assign || function (target) {
  for (var i = 1; i < arguments.length; i++) {
    var source = arguments[i];

    for (var key in source) {
      if (Object.prototype.hasOwnProperty.call(source, key)) {
        target[key] = source[key];
      }
    }
  }

  return target;
};

var inherits = function (subClass, superClass) {
  if (typeof superClass !== "function" && superClass !== null) {
    throw new TypeError("Super expression must either be null or a function, not " + typeof superClass);
  }

  subClass.prototype = Object.create(superClass && superClass.prototype, {
    constructor: {
      value: subClass,
      enumerable: false,
      writable: true,
      configurable: true
    }
  });
  if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass;
};

var possibleConstructorReturn = function (self, call) {
  if (!self) {
    throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
  }

  return call && (typeof call === "object" || typeof call === "function") ? call : self;
};

var ReactCountryFlag = function (_Component) {
    inherits(ReactCountryFlag, _Component);

    function ReactCountryFlag() {
        classCallCheck(this, ReactCountryFlag);
        return possibleConstructorReturn(this, (ReactCountryFlag.__proto__ || Object.getPrototypeOf(ReactCountryFlag)).apply(this, arguments));
    }

    createClass(ReactCountryFlag, [{
        key: "render",
        value: function render() {
            var _props = this.props,
                cdnUrl = _props.cdnUrl,
                code = _props.code,
                styleProps = _props.styleProps,
                svg = _props.svg,
                title = _props.title;


            var flagUrl = "" + cdnUrl + code.toLowerCase() + ".svg";
            var emoji = code.toUpperCase().replace(/./g, function (char) {
                return String.fromCodePoint(char.charCodeAt(0) + 127397);
            });

            return svg ? React__default.createElement("span", {
                "aria-label": code,
                role: "img",
                style: _extends({
                    position: "relative",
                    display: "inline-block",
                    width: "1.3333333333em",
                    height: "1em",
                    backgroundImage: "url(" + flagUrl + ")",
                    backgroundPosition: "50%",
                    backgroundRepeat: "no-repeat",
                    backgroundSize: "contain",
                    fontSize: "1em",
                    lineHeight: "1em",
                    verticalAlign: "middle"
                }, styleProps),
                title: title || code
            }) : React__default.createElement(
                "span",
                {
                    "aria-label": code,
                    role: "img",
                    style: _extends({
                        verticalAlign: "middle"
                    }, styleProps),
                    title: code
                },
                emoji
            );
        }
    }]);
    return ReactCountryFlag;
}(React.Component);

ReactCountryFlag.propTypes = {
    cdnUrl: PropTypes.string,
    code: PropTypes.string.isRequired,
    styleProps: PropTypes.object,
    svg: PropTypes.bool,
    title: PropTypes.string
};
ReactCountryFlag.defaultProps = {
    cdnUrl: "https://cdnjs.cloudflare.com/ajax/libs/flag-icon-css/3.2.1/flags/4x3/",
    svg: false
};

module.exports = ReactCountryFlag;
